### PR TITLE
fix: 1116 silent error on sell order creation

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1116

- fix silent error on failed create sell order transaction
- simplify error modal conditional display
- change error button label on error modal for failed create sell order transaction
- fix console error on incorrect SVG properties
- fix console error on nested `p` tag

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

### How to test

1. Go to https://deploy-preview-1132--regen-registry.netlify.app/
2. Try to. create a sell order
3. The error popup should be displayed because the preview is still targeting v4 testnet

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/src/components/icons/AvailableCreditsIconAlt.tsx [hunk 1]
```diff
@@ -21,8 +21,8 @@ export default function AvailableCreditsIcon({
         xmlns="http://www.w3.org/2000/svg"
       >
         <path
-          fill-rule="evenodd"
-          clip-rule="evenodd"
+          fillRule="evenodd"
+          clipRule="evenodd"
```
Comment (flagrede) [inline web-components/src/components/icons/AvailableCreditsIconAlt.tsx:25]: change to valid React properties

File: web-components/src/components/inputs/CheckboxLabel.tsx [hunk 2]
```diff
@@ -42,7 +44,7 @@ export default function CheckboxLabel({
             disabled={disabled}
           />
         }
-        label={<Subtitle>{label}</Subtitle>}
+        label={isLabelStringOrNumber ? <Subtitle>{label}</Subtitle> : label}
```
Comment (flagrede) [inline web-components/src/components/inputs/CheckboxLabel.tsx:47]: Prevent the use of nested typography components.
We were having an error related to that before.

File: web-registry/src/pages/Dashboard/MyEcocredits/MyEcocredits.tsx [hunk 3]
```diff
@@ -100,7 +101,7 @@ export const MyEcocredits = (): JSX.Element => {
 
   const onButtonClick = (): void => {
     handleTxModalClose();
-    if (txButtonTitle === CREATE_SELL_ORDER_BUTTON) {
+    if (txButtonTitle === CREATE_SELL_ORDER_BUTTON && !error) {
```
Comment (flagrede) [inline web-registry/src/pages/Dashboard/MyEcocredits/MyEcocredits.tsx:104]: Do not redirect to the sell order table on error modal button click.

File: web-registry/src/pages/Dashboard/MyEcocredits/MyEcocredits.tsx [hunk 4]
```diff
@@ -397,19 +398,17 @@ export const MyEcocredits = (): JSX.Element => {
...
-          txHashUrl={txHashUrl}
-          cardTitle={txModalTitle ?? ''}
-          buttonTitle={txButtonTitle}
-          linkComponent={Link}
-          onButtonClick={onButtonClick}
-        />
-      )}
+      <TxErrorModal
+        error={error ?? ''}
+        open={!!error}
```
Comment (flagrede) [inline web-registry/src/pages/Dashboard/MyEcocredits/MyEcocredits.tsx:403]: Simplify error modal conditional display.

File: web-registry/src/pages/Dashboard/MyEcocredits/MyEcocredits.tsx [hunk 5]
```diff
@@ -397,19 +398,17 @@ export const MyEcocredits = (): JSX.Element => {
...
+        onClose={handleTxModalClose}
+        txHash={txHash || ''}
+        txHashUrl={txHashUrl}
+        cardTitle={txModalTitle ?? ''}
+        buttonTitle={ERROR_BUTTON}
```
Comment (flagrede) [inline web-registry/src/pages/Dashboard/MyEcocredits/MyEcocredits.tsx:408]: Use the same error button title for all actions.

File: web-registry/src/pages/Dashboard/MyEcocredits/hooks/useCreateSellOrderSubmit.tsx [hunk 6]
```diff
@@ -37,7 +37,7 @@ const useCreateSellOrderSubmit = ({
   setTxButtonTitle,
   setSellOrderCreateOpen,
 }: Props): ReturnType => {
-  const basketPutSubmit = useCallback(
+  const createSellOrderSubmit = useCallback(
```
Comment (flagrede) [inline web-registry/src/pages/Dashboard/MyEcocredits/hooks/useCreateSellOrderSubmit.tsx:40]: fix typo.

**Comment by flagrede** [comment]
@erikalogie You should be able to see the error modal now when trying to create a sell order. The preview still target the v4 testnet and has an error on that transaction at the moment (not sure why) but you can check there. Everything is fine on redwood.

**Comment by erikalogie** [comment]
@flagrede I don't have any ecocredits on production. Is there another place I can test this?

**Comment by erikalogie** [comment]
@flagrede I tried this on https://deploy-preview-1132--regen-registry.netlify.app/ and it appears to be working!

**Comment by flagrede** [comment]
> @flagrede I don't have any ecocredits on production. Is there another place I can test this?

Sorry I paste the wrong link in the description, I meant to paste the preview one.

File: web-components/src/components/icons/AvailableCreditsIconAlt.tsx (hunk 1 above)
Comment (mhagel) [reply web-components/src/components/icons/AvailableCreditsIconAlt.tsx:25]: thanks, these are easy to miss sometimes

**Review by mhagel — APPROVED** [review]

**Review by victordelval — APPROVED** [review]

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**